File: 3712-sum-of-elements-with-frequency-divisible-by-k/3712-sum-of-elements-with-frequency-divisible-by-k.java
class Solution {
    public int sumDivisibleByK(int[] nums, int k) {
        Map<Integer,Integer> map= new HashMap<>();
        for(Integer t : nums)
        map.merge(t,1,Integer::sum);
        int s = 0;
        for(Map.Entry<Integer,Integer> e : map.entrySet())
        if(e.getValue() % k == 0)
        s = s + (e.getKey() * e.getValue());
        return s;

        
    }
}

File: 3712-sum-of-elements-with-frequency-divisible-by-k/3712-sum-of-elements-with-frequency-divisible-by-k.c
int sumDivisibleByK(int* nums, int numsSize, int k) {
    int i,j,c,s=0,f;
    for(i=0;i<numsSize;i++)
    {
        f=0;
        for(j=0;j<i;j++)
        if(nums[i]==nums[j])
        {
            f=1;
            break;
        }
        if(f==0)
        {
        c=0;
        for(j=i;j<numsSize;j++)
        if(nums[i]==nums[j])
        c++;
        if(c%k==0)
        s=s+(c*nums[i]);
        }
    }
    return s;
}